Python/pylifecycle.c

```diff
@@ -1910,6 +1909,9 @@ finalize_interp_clear(PyThreadState *tstate)
 
     finalize_interp_types(tstate->interp);
 
+    /* Finalize dtoa at last so that finalizers calling repr of float doesn't crash */
+    _PyDtoa_Fini(tstate->interp);
```

Comment:
Does the function actually rely on `PyInterpreterState`?  If so, I'd expect moving this after `_PyInterpreterState_Clear()` would be a problem.  Otherwise, I would expect `_PyDtoa_Fini()` to not take any arguments.

Reply:
dtoa state is stored in PyInterpreterState. I don't see how it's an issue to call _PyDtoa_Fini() after finalize_interp_types(), since _PyDtoa_Fini() doesn't use Python objects, it only clears its state.

Reply:
Yes, Victor is right, it clears interpreter specific data hence needs the argument to interp state.